File: epozoriste_mobile/lib/models/kupovina.dart
import 'package:epozoriste_mobile/models/models.dart';
import 'package:json_annotation/json_annotation.dart';

part 'kupovina.g.dart';

@JsonSerializable()
class Kupovina {
  int kupovinaId;
  int? kolicina;
  int? cijena;
  DateTime datumKupovine;
  int? korisnikId;
  Korisnik? korisnik;

  Kupovina({
    required this.kupovinaId,
    this.kolicina,
    this.cijena,
    required this.datumKupovine,
    this.korisnikId,
    this.korisnik,
  });

  factory Kupovina.fromJson(Map<String, dynamic> json) =>
      _$KupovinaFromJson(json);
  Map<String, dynamic> toJson() => _$KupovinaToJson(this);
}

File: ePozoriste.Model/Kupovina.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ePozoriste.Model
{
    public class Kupovina
    {
        public int KupovinaId { get; set; }
        public int? Kolicina { get; set; }
        public int? Cijena { get; set; }
        public DateTime? DatumKupovine { get; set; }
        public int? KorisnikId { get; set; }
        public virtual Korisnik? Korisnik { get; set; }
    }
}
